a11y-dialog.js

```diff
@@ -443,7 +443,7 @@
   if (typeof module !== 'undefined' && typeof module.exports !== 'undefined') {
     module.exports = A11yDialog;
   } else if (typeof define === 'function' && define.amd) {
-    define('A11yDialog', [], function() {
+    define([], function() {
```

Comment:
`A11yDialog` here is not a dependency but the name of the module. See https://requirejs.org/docs/whyamd.html#namedmodules.

However this change still makes sense for others reasons:
> Notice that the above module does not declare a name for itself. This is what makes the module very portable. It allows a developer to place the module in a different path to give it a different ID/name. The AMD loader will give the module an ID based on how it is referenced by other scripts.
>
> However, tools that combine multiple modules together for performance need a way to give names to each module in the optimized file. For that, AMD allows a string as the first argument to define():

@edenspiekermann I would recommend you to split the source code and logics from the module definition. Define a simple ES6 module and build from it AMD/CJS/ESM/UMD modules with packages like [rollup](https://github.com/rollup/rollup) or [parcel](https://github.com/parcel-bundler/parcel).

Reply:
I’ll see what I can do, thanks for your input @ncoden.

Reply:
In my (legacy) project I'm not using a build tool. The project is running Require JS. Problem in my case was that I used an other path name 'dialog'. It works when I use 'A11yDialog' like defined in the code.

So it was not a circular dependency issue but rather a routing/loading issue. Still think it's better not to name the AMD module.

Reply:
If we’re being honest, I don‘t really want to start change the build step for this library… Wouldn’t this PR simply fix the problem?

Reply:
Yes it would. I can understand you don't want to change the build process. I'm just saying what would be the best: leaving browser support, bundling and delivering concerns to the specialised packages.